SU2_CFD/include/variables/CFlowVariable.hpp

```diff
@@ -0,0 +1,226 @@
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "CVariable.hpp"
+
+/*!
+ * \class CFlowVariable
+ * \brief Class for defining the common variables of flow solvers.
+ */
+class CFlowVariable : public CVariable {
+ protected:
+  /*--- Primitive variable definition. ---*/
+  MatrixType Primitive;                     /*!< \brief Primitive variables. */
+  CVectorOfMatrix Gradient_Primitive;       /*!< \brief Gradient of the primitive variables. */
+  CVectorOfMatrix& Gradient_Reconstruction; /*!< \brief Reference to the gradient of the primitive variables for MUSCL reconstruction for the convective term */
+  CVectorOfMatrix Gradient_Aux;             /*!< \brief Auxiliary structure to store a second gradient for reconstruction, if required. */
+  MatrixType Limiter_Primitive;             /*!< \brief Limiter of the primitive variables. */
+  VectorType Velocity2;                     /*!< \brief Squared norm of velocity. */
+
+  MatrixType Solution_New;    /*!< \brief New solution container for Classical RK4. */
+  MatrixType HB_Source;       /*!< \brief harmonic balance source term. */
+
+  /*--- NS Variables declared here to make it easier to re-use code between compressible and incompressible solvers. ---*/
+  MatrixType Vorticity;       /*!< \brief Vorticity of the fluid. */
+  VectorType StrainMag;       /*!< \brief Magnitude of rate of strain tensor. */
+
+  /*!
+   * \brief Constructor of the class.
+   * \note This class is not meant to be instantiated directly, it is only a building block.
+   * \param[in] npoint - Number of points/nodes/vertices in the domain.
+   * \param[in] ndim - Number of dimensions.
+   * \param[in] nvar - Number of variables.
+   * \param[in] nprimvar - Number of primitive variables.
+   * \param[in] nprimvargrad - Number of primitive variables for which to compute gradients.
+   * \param[in] config - Definition of the particular problem.
+   */
+  CFlowVariable(unsigned long npoint, unsigned long ndim, unsigned long nvar,
+                unsigned long nprimvar, unsigned long nprimvargrad, const CConfig *config);
+
+ public:
+  /*!
+   * \brief Get a primitive variable.
+   * \param[in] iVar - Index of the variable.
+   * \return Value of the primitive variable for the index <i>iVar</i>.
+   */
+  inline su2double GetPrimitive(unsigned long iPoint, unsigned long iVar) const final { return Primitive(iPoint,iVar); }
+
+  /*!
+   * \brief Get all the primitive variables of the problem.
+   * \return Pointer to the primitive variable vector.
+   */
+  inline su2double *GetPrimitive(unsigned long iPoint) final { return Primitive[iPoint]; }
```

Comment:
I think you did not run clang-format over this the latest file 🤨  `pointerAlignement: left` is my reason here :)